src/classes/HouseholdAccounts.cls

```diff
@@ -98,4 +157,362 @@ public inherited sharing class HouseholdAccounts {
+                WHERE AccountId IN :accountIds
+        ]) {
+            Set<Id> contactIds = contactIdsByAccountId.get(contactRecord.AccountId);
+            if (contactIds == null) {
+                contactIds = new Set<Id>();
+                contactIdsByAccountId.put(contactRecord.AccountId, contactIds);
+            }
+            contactIds.add(contactRecord.Id);
+        }
+
+        // now go thru each account and update their primary contact if the current one isn't valid
+        List<Account> accounts = new List<Account>();
+        for (Account accountRecord : [SELECT Id, npe01__One2OneContact__c FROM Account WHERE Id IN :accountIds]) {
+            Set<Id> contactIds = contactIdsByAccountId.get(accountRecord.Id);
+            if (contactIds != null && !contactIds.contains(accountRecord.npe01__One2OneContact__c)) {
+                accountRecord.npe01__One2OneContact__c = extractFirstId(contactIds);
+                accounts.add(accountRecord);
+            }
+        }
+
+        if (accounts != null && !accounts.isEmpty()) {
+            dmlWrapper.objectsToUpdate.addAll(accounts);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description returns the first Id from the specified set
+    * @param ids a set of Id's
+    * @return Id.  null if the set is empty.
+    */
+    private static Id extractFirstId(Set<Id> ids) {
+        for (Id recordId : ids) {
+            return recordId;
+        }
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of AccountId's, recalc their rollups
+    * @param accountIds the list of Account (Id's) that need updating.
+    */
+    public static void rollupAccounts(List<Id> accountIds) {
+        new RLLP_OppRollup().rollupAccounts(new Set<Id>(accountIds));
+    }
+
+    /*******************************************************************************************************
+    * @description Inserts a new 1:1 or HH Account for an Individual Contact
+    * @param contactsWithoutAccounts the list of Contacts that need updating.
+    */
+    public static void createOneToOneAndHouseholdAccountsFor(List<Contact>
+            contactsWithoutAccounts){
+        if (contactsWithoutAccounts.isEmpty()) {
+            return;
+        }
+
+        List<Id> contactIds = new List<Id>();
+        Map<Id, Account> accountByPrimaryContactId = new Map<Id, Account>();
+        List<Contact> contacts = new List<Contact>();
+
+        for (Contact contactRecord : contactsWithoutAccounts) {
+            //track which contacts are being inserted vs. updated
+            if (contactRecord.Id != null) {
+                contactIds.add(contactRecord.Id);
+            }
+        }
+        // get all the Accounts that are connected to the existing Contacts
+        for (Account accountRecord : [
```

Comment:
More queries, maybe this is a larger problem.

Reply:
This method was fully lifted from ACCT_IndividualAccounts_TDTM, and HouseholdAccounts is evolving from its original incarnation as a way to lazy load a Households' members.  

Now that we have these methods in the same domain, we should absolutely be able to take advantage and query what is needed up front at instantiation, so this type of multiple query pattern can be avoided.

I'll take a look today and see if this seems to make sense to do now.

Reply:
I really like these ideas about combining queries in this class and getting it instantiated with the right set of records.  But going to hold off on that for a follow up next WI: https://salesforce.quip.com/n2vzAQOiRbT7#UMTACAR9Raq